Formula/percona-xtrabackup.rb

```diff
@@ -0,0 +1,56 @@
+
+class PerconaXtrabackup < Formula
+  desc "Open source hot backup tool for InnoDB and XtraDB databases"
+  homepage "https://www.percona.com/software/mysql-database/percona-xtrabackup"
+  url "https://www.percona.com/downloads/XtraBackup/Percona-XtraBackup-2.4.2/source/tarball/percona-xtrabackup-2.4.2.tar.gz"
+  sha256 "faeac6f1db4a1270e5263e48c8a94cc5c81c772fdea36879d1be18dcbcd1926e"
+
+  option "with-man-pages", "Build man pages. Needs python-sphinx to be installed"
+
+  depends_on "cmake" => :build
+  depends_on "libev" => :build
+  depends_on "libgcrypt"
+  depends_on :mysql => :recommended
+
+  def install
+    cmake_args = %W[
+      -DBUILD_CONFIG=xtrabackup_release
+      -DDOWNLOAD_BOOST=1
```

Comment:
> I noticed this commit in a legacy mysql formula: Homebrew/legacy-homebrew@faef69a

This method is the one we prefer.

>  it's really similar to letting the build download it, but with the disadvantage that the version is hardcoded here instead of being automatically resolved. In that sense, I'd think it's better to leave the downloading to the build.

It's similar but we try to not intentionally let the _build process itself_ do remote code fetching.

Partly because we can't guarantee what security measures upstream is taking, whether they're using their own servers or upstreams, whether they do any sort of checksum verification, etc, and in part because it means you can't install the formula offline, which is something we endeavour to support where and when possible.